src/signals/incident/services/check-visibility/index.test.js

```diff
@@ -0,0 +1,199 @@
+      control.meta.ifAllOf = {
+        subcategory: 'wrong'
+      };
+      expect(checkVisibility(control, incident)).toBeFalsy();
+    });
+
+    it('array: should be visible when category is valid', () => {
+      control.meta.ifAllOf = {
+        category: ['bar']
+      };
+      expect(checkVisibility(control, incident)).toBeTruthy();
+    });
+
+    it('array: should not be visible when category is not valid', () => {
+      control.meta.ifAllOf = {
+        category: ['bar', 'wrong']
+      };
+      expect(checkVisibility(control, incident)).toBeFalsy();
+    });
+
+    it('object with id: should be visible when object_with_id is valid', () => {
+      control.meta.ifAllOf = {
+        object_with_id: 'id'
+      };
+      expect(checkVisibility(control, incident)).toBeTruthy();
+    });
+
+    it('object with id: should not be visible when object_with_id is not valid', () => {
+      control.meta.ifAllOf = {
+        object_with_id: 'wrong'
+      };
+      expect(checkVisibility(control, incident)).toBeFalsy();
+    });
+
+    it('object with value: should be visible when object_with_value is valid', () => {
+      control.meta.ifAllOf = {
+        object_with_value: 'value'
+      };
+      expect(checkVisibility(control, incident)).toBeTruthy();
+    });
+
+    it('object with value: should not be visible when object_with_value is not valid', () => {
+      control.meta.ifAllOf = {
+        object_with_value: 'wrong'
+      };
+      expect(checkVisibility(control, incident)).toBeFalsy();
+    });
+
+    it('array with objects with id: should be visible when array_with_object_with_id is valid', () => {
+      control.meta.ifAllOf = {
+        array_with_object_with_id: 'id'
+      };
+      expect(checkVisibility(control, incident)).toBeTruthy();
+    });
+
+    it('array with objects with id: should not be visible when array_with_object_with_id is not valid', () => {
+      control.meta.ifAllOf = {
+        array_with_object_with_id: 'wrong'
+      };
+      expect(checkVisibility(control, incident)).toBeFalsy();
+    });
+  });
+
+  describe('ifOneOf', () => {
+    it('string: should be visible when category is valid', () => {
+      control.meta.ifOneOf = {
+        category: 'bar'
+      };
+      expect(checkVisibility(control, incident)).toBeTruthy();
+    });
+
+    it('string: should not be visible when subcategory is not valid', () => {
+      control.meta.ifOneOf = {
+        subcategory: 'wrong'
+      };
+      expect(checkVisibility(control, incident)).toBeFalsy();
+    });
+
+    it('string: should be visible when color is one of array defined', () => {
+      control.meta.ifOneOf = {
+        color: 'blue'
+      };
+      expect(checkVisibility(control, incident)).toBeTruthy();
+    });
+
+    it('string: should not be visible when color is not one of array defined', () => {
+      control.meta.ifOneOf = {
+        color: 'wrongcolor'
+      };
+      expect(checkVisibility(control, incident)).toBeFalsy();
```

Comment:
`.toBe(false)` is a lot stronger statement. `toBeFalsy` isn't something you generally want.

Reply:
ja klopt! ik verander het overal